hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -627,23 +633,9 @@ private double calculateAvgUtilization(List<DatanodeUsageInfo> nodes) {
     }
     clusterCapacity = aggregatedStats.getCapacity().get();
     clusterUsed = aggregatedStats.getScmUsed().get();
+    clusterRemaining = aggregatedStats.getRemaining().get();
 
-    return clusterUsed / (double) clusterCapacity;
-  }
-
-  /**
-   * Calculates the utilization, that is used space divided by capacity, for
-   * the given datanodeUsageInfo.
-   *
-   * @param datanodeUsageInfo DatanodeUsageInfo to calculate utilization for
-   * @return Utilization value
-   */
-  public static double calculateUtilization(
-      DatanodeUsageInfo datanodeUsageInfo) {
-    SCMNodeStat stat = datanodeUsageInfo.getScmNodeStat();
-
-    return stat.getScmUsed().get().doubleValue() /
-        stat.getCapacity().get().doubleValue();
+    return (clusterCapacity - clusterRemaining) / (double) clusterCapacity;
```

Comment:
when calculating the average utilization, should we use the average value of the utilizations of every datanode?

Reply:
I think both these methods will work well for us. Did you notice any problems with the current implementation?

Reply:
both methods are acceptable to me, and i notice HDFS balancer uses the same method as current implementation, so let us keep the same with HDFS.